Mathlib/RingTheory/Valuation/ValuationSubring.lean

```diff
@@ -77,6 +79,13 @@ theorem neg_mem (x : K) : x ∈ A → -x ∈ A := A.toSubring.neg_mem
 theorem mem_or_inv_mem (x : K) : x ∈ A ∨ x⁻¹ ∈ A := A.mem_or_inv_mem' _
 #align valuation_subring.mem_or_inv_mem ValuationSubring.mem_or_inv_mem
 
+theorem Place.integralClosure_le {R} [Field R] [Algebra R K] (v : Place K R) :
+    integralClosure R K ≤ v.toSubalgebra := by
+  intro z hz
+  by_contra hzv
+  have : z⁻¹ ∈ v.toSubalgebra := (mem_or_inv_mem v.toValuationSubring z).resolve_left hzv
+  exact hzv (IsIntegral.mem_of_inv_mem hz this)
```

Comment:
I've been able to show this @MichaelStollBayreuth

Reply:
Very nice!